DuckDuckGo/UserBehaviorMonitor.swift

```diff
@@ -64,32 +71,46 @@ final class UserBehaviorMonitor {
+    var didDoubleRefreshTimestamp: Date? {
+        get { store.didDoubleRefreshTimestamp }
+        set { store.didDoubleRefreshTimestamp = newValue }
+    }
+
+    var didRefreshCounter: Int {
+        get { store.didRefreshCounter }
+        set { store.didRefreshCounter = newValue }
     }
 
     func handleAction(_ action: Action, date: Date = Date()) {
         switch action {
         case .refresh:
-            fireEventIfActionOccurredRecently(since: didRefreshTimestamp, eventToFire: .reloadTwice, within: 10.0)
+            fireEventIfActionOccurredRecently(within: 12.0, since: didRefreshTimestamp, eventToFire: .reloadTwiceWithin12Seconds)
+            fireEventIfActionOccurredRecently(within: 24.0, since: didRefreshTimestamp, eventToFire: .reloadTwiceWithin24Seconds)
             didRefreshTimestamp = date
-        case .burn:
-            fireEventIfActionOccurredRecently(since: didRefreshTimestamp, eventToFire: .reloadAndFireButton)
-            didBurnTimestamp = date
+            
+            if didRefreshCounter == 0 {
+                didDoubleRefreshTimestamp = date
+            }
+            didRefreshCounter += 1
+            if didRefreshCounter > 2 {
+                fireEventIfActionOccurredRecently(within: 20.0, since: didDoubleRefreshTimestamp, eventToFire: .reloadThreeTimesWithin20Seconds)
+                fireEventIfActionOccurredRecently(within: 40.0, since: didDoubleRefreshTimestamp, eventToFire: .reloadThreeTimesWithin40Seconds)
+                didRefreshCounter = 0
+            }
         case .reopenApp:
-            fireEventIfActionOccurredRecently(since: didRefreshTimestamp, eventToFire: .reloadAndRestart)
-            fireEventIfActionOccurredRecently(since: didBurnTimestamp, eventToFire: .fireButtonAndRestart)
-        case .openSettings:
-            fireEventIfActionOccurredRecently(since: didRefreshTimestamp, eventToFire: .reloadAndOpenSettings)
-        case .toggleProtections:
-            fireEventIfActionOccurredRecently(since: didRefreshTimestamp, eventToFire: .reloadAndTogglePrivacyControls)
-            fireEventIfActionOccurredRecently(since: didBurnTimestamp, eventToFire: .fireButtonAndTogglePrivacyControls)
+            fireEventIfActionOccurredRecently(within: 30.0, since: didRefreshTimestamp, eventToFire: .reloadAndRestartWithin30Seconds)
+            fireEventIfActionOccurredRecently(within: 50.0, since: didRefreshTimestamp, eventToFire: .reloadAndRestartWithin50Seconds)
         }
 
-        func fireEventIfActionOccurredRecently(since timestamp: Date?, eventToFire: UserBehaviorEvent, within interval: Double = 30.0) {
+        func fireEventIfActionOccurredRecently(within interval: Double = 30.0, since timestamp: Date?, eventToFire: UserBehaviorEvent) {
             if let timestamp = timestamp, date.timeIntervalSince(timestamp) < interval {
                 eventMapping.fire(eventToFire)
+                PixelExperiment.install() // Do we have better place to install it?
```

Comment:
This should not be called here - there is a line commented out in the AppDelegate that is responsible for this which will be commented in when @tomasstrba 's experiment goes live.  Cohorts are allocated at the point you access the `.cohort` variable.